plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -264,44 +265,49 @@ public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStati
-                    .map(columnChunk ->
-                            runAsync(() -> glueClient.updateColumnStatisticsForPartition(new UpdateColumnStatisticsForPartitionRequest()
+            columnChunks.forEach(columnChunk ->
+                    updateFutures.add(runAsync(() -> glueClient.updateColumnStatisticsForPartition(
+                            new UpdateColumnStatisticsForPartitionRequest()
                                     .withCatalogId(catalogId)
                                     .withDatabaseName(partition.getDatabaseName())
                                     .withTableName(partition.getTableName())
                                     .withPartitionValues(partition.getValues())
-                                    .withColumnStatisticsList(columnChunk)), writeExecutor))
-                    .collect(toUnmodifiableList());
+                                    .withColumnStatisticsList(columnChunk)),
+                            writeExecutor)));
 
-            Map<String, HiveColumnStatistics> currentColumnStatistics = this.getPartitionColumnStatisticsIfPresent(partition).orElse(ImmutableMap.of());
-            Set<String> removedStatistics = difference(currentColumnStatistics.keySet(), updatedColumnStatistics.keySet());
-            List<CompletableFuture<Void>> deleteStatsFutures = removedStatistics.stream()
-                    .map(column -> runAsync(() ->
-                            glueClient.deleteColumnStatisticsForPartition(new DeleteColumnStatisticsForPartitionRequest()
+            Set<String> removedStatistics = difference(currentStatistics.get(partition).keySet(), updatedColumnStatistics.keySet());
+            removedStatistics.forEach(column ->
+                    updateFutures.add(runAsync(() -> glueClient.deleteColumnStatisticsForPartition(
+                            new DeleteColumnStatisticsForPartitionRequest()
                                     .withCatalogId(catalogId)
                                     .withDatabaseName(partition.getDatabaseName())
                                     .withTableName(partition.getTableName())
                                     .withPartitionValues(partition.getValues())
-                                    .withColumnName(column)), writeExecutor))
-                    .collect(toUnmodifiableList());
-
-            ImmutableList<CompletableFuture<Void>> updateOperationsFutures = ImmutableList.<CompletableFuture<Void>>builder()
-                    .addAll(writePartitionStatsFutures)
-                    .addAll(deleteStatsFutures)
-                    .build();
-
-            getFutureValue(allOf(updateOperationsFutures.toArray(CompletableFuture[]::new)));
+                                    .withColumnName(column)),
+                            writeExecutor)));
+        }
+        try {
+            getFutureValue(allOf(updateFutures.toArray(CompletableFuture[]::new)));
+        }
+        catch (EntityNotFoundException ex) {
```

Comment:
I'm not sure if `getFutureValue` returns the original exception unwrapped. You might need to catch the RTE and then unwrap it to see the underlying cause.

Reply:
You're right, all exceptions are wrapped in `RuntimeException(cause)`, changed to check the cause for EntityNotFound